src/node_os.cc

```diff
@@ -365,38 +366,31 @@ static void GetUserInfo(const FunctionCallbackInfo<Value>& args) {
                                                    pwd.username,
                                                    encoding,
                                                    &error);
+  if (username.IsEmpty()) {
+  fail_with_stringbytes_error:
+    uv_os_free_passwd(&pwd);
+    env->isolate()->ThrowException(error);
+    return;
+  }
```

Comment:
`error` is cleared by `StringBytes::Encode()` if the call succeeded… I’ve thought about doing that, would you prefer to change that behaviour? I can see the advantage that would have.